hips/hip-9999.md

```diff
@@ -0,0 +1,170 @@
+---
+hip: 9999
+title: "Helm 4 Development Process"
+authors: [ "Matt Butcher <matt.butcher@microsoft.com>" ]
+created: "2020-08-13"
+type: "process"
+status: "draft"
+---
+
+## Abstract
+
+The Helm 4 development lifecycle will be managed by a release engineer.
+It will be kicked off in October, 2021.
+The process will last not less than six months.
+There will be an active development phase, followed by an alpha period, a beta period, and a release candidate period.
+The release manager will determine when Helm 4 is ready to ship.
+
+## Motivation
+
+We do not currently have a documented process for kickstarting new major Helm versions.
+During the Helm 3 process, we drew criticism for not having such documentation.
+However, a "generalized" process that can span multiple major versions over multiple
+years is ambitious.
+So this proposal attempts to merely spell out a formal process for Helm 4.
+
+## Rationale
+
+Describe why particular design decisions were made.
+
+## Specification
+
+### The Release Engineer
+
+A single individual will oversee the planning and development phases of Helm 4.
+This person will be the _Release Engineer_.
+The person will have the following responsibilities:
+
+- Lead the kick-off meeting
+- Define the process for including features
+- Determine the timelines for development
+- Make the determination of when the project has indeed reached a milestone
+- Name and oversee releases
+- After the release, the release manager will determine what the "intent" of a Helm behavior was (which informs determining whether an incoming breaking change is a feature or a bugfix)
+
+#### Determining Who Will Be Release Engineer
+
+The release engineer must be a core maintainer on the `github.com/helm/helm` project.
+The engineer must not be in danger of losing core maintainership due to inactivity.
+Anyone can nominate a core maintainer to become the release engineer (self-nominations are encouraged).
+Any nominee must _agree_ to the nomination.
+The Helm org maintainers will determine which of the nominated maintainers shall be the release engineer.
```

Comment:
Interesting choice here. Shouldn't the core maintainers of a project come to a consensus on their own by default? If there is a conflict, then it can be bubbled up to the org maintainers as per governance.

Reply:
I would be okay with that. I think you are right that we tend not to appeal to org maintainers if we don't have to